### PR TITLE
Add indexing workflow

### PR DESCRIPTION
**Pull request by dshevtsov**
## Description

Adding a workflow to run indexing with the following options:
- manually
- by schedule (weekly, on Saturdays)

## Related Issue

- https://github.com/AdobeDocs/commerce-php/pull/112
- https://github.com/AdobeDocs/commerce-admin-developer/pull/44
- https://github.com/AdobeDocs/commerce-cloud-tools/pull/35
- https://github.com/AdobeDocs/commerce-contributor/pull/65
- https://github.com/AdobeDocs/commerce-events/pull/99
- https://github.com/AdobeDocs/commerce-frontend-core/pull/86
- https://github.com/AdobeDocs/commerce-marketplace/pull/70
- https://github.com/AdobeDocs/commerce-pwa-studio/pull/166
- https://github.com/AdobeDocs/commerce-services/pull/38
- https://github.com/AdobeDocs/commerce-testing/pull/32
- https://github.com/AdobeDocs/commerce-webapi/pull/96
- https://github.com/AdobeDocs/graphql-mesh-gateway/pull/102

## Motivation and Context

Proper indexing automation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/index.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: Indexing
+on:
+  schedule:
+    - cron: '42 20 * * SAT'
```
Comment (jeff-matthews) [inline .github/workflows/index.yml:5]: Remember to change weekly cron to monthly, especially for repos that don't get a lot of changes.

**Comment by dshevtsov** [comment]
@jeff-matthews, please add an org secret `AIO_ALGOLIA_WRITE_API_KEY` for this project.

**Review by jeff-matthews — COMMENTED** [review]
FYYI - This repo hasn't been indexed before. At least, I don't see a `commerce-xd-kits` index in Algolia.

File: .github/workflows/index.yml (hunk 1 above)
Comment (dshevtsov) [reply .github/workflows/index.yml:5]: It's up-to project owners. I'm just adding a "template" workflow.

**Comment by dshevtsov** [comment]
> FYYI - This repo hasn't been indexed before. At least, I don't see a `commerce-xd-kits` index in Algolia.

I'm adding indexing because it's in [search-indices|https://github.com/AdobeDocs/search-indices/blob/main/product-index-map.json#L102].

**Comment by jeff-matthews** [comment]
> @jeff-matthews, please add an org secret `AIO_ALGOLIA_WRITE_API_KEY` for this project.

Done @dshevtsov

File: .github/workflows/index.yml (hunk 1 above)
Comment (dshevtsov) [reply .github/workflows/index.yml]: Changed to monthly.

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**